### DISCUSSION TITLE
Styling Navigator container

### DISCUSSION THREAD

**Discussion by jonathanroze** [react-navigation/react-navigation#9431] [Q&A]
Hi, 

I'm using react-navigation 5.

I want to have a Stack container with rounded border and I tried different way to doing that but I can have great result.

I also tried by using custom navigator and it look fine but I loosing transitions between pages and header. And it seems to be a very hard work to implement all of this manually. 

So do you have some idea to do this ? 

Here a picture of wanted result.

Thanks

@react-navigation/native: 5.8.10
@react-navigation/stack 5.12.8

<img width="480" alt="Capture d’écran 2021-03-18 à 08 44 31" src="https://user-images.githubusercontent.com/2753799/111590284-348f1a00-87c6-11eb-9afe-807b61fa2f0a.png">

**Comment by raajnadar** [comment]
There are many styling options to customise the stack navigator https://reactnavigation.org/docs/stack-navigator/#options

Edit: - You can try https://reactnavigation.org/docs/modal/ and design the modal as per your needs.

**Reply by jonathanroze** [reply]
Hi @raajnadar, thanks for you reply but I can find any way to do this result with props exposed

**Reply by jonathanroze** [reply]
Is there a way to customize StackView without rewrite all of the component (animations etc..) ? (Just customize container)

**Reply by raajnadar** [reply]
I did the same design with the default stack navigation. I used 2 nested views the inner view with marginTop. Or else you can use position absolute.

**Reply by jonathanroze** [reply]
Thank for your feedback, but this way, when you navigate to another page, this view will slide too, it not pretty beautiful

**Reply by raajnadar** [reply]
IDK what are you expecting, looking at your attached image it is very easy design in react native. I didn't understand what issue you are facing due to the navigation library.

**Reply by jonathanroze** [reply]
I'm sorry I'm maybe not very clean. I will try to explain better my need.

I need the navigator container (where the screens are displayed) to be rounded on the top edges. And that the animations play in this space. I managed to have rounded edges in one screen but if I change the page, the block with the rounded edges slides too. And I would like the animation to be inside this container with the rounded edges. I hope this is clearer?

**Reply by raajnadar** [reply]
You can try https://reactnavigation.org/docs/modal/ and design the modal as per your needs.
